cli/cmd/encore/gen.go

```diff
@@ -37,15 +44,29 @@ Use '--env=local' to generate it based on your local development version of the
 Supported language codes are:
   typescript: A TypeScript-client using the in-browser Fetch API
   go: A Go client using net/http"
+
+Support presets are:
+	nextjs: Generates a TypScript client for use within a Next.js application
 `,
 		Args: cobra.ExactArgs(1),
 		Run: func(cmd *cobra.Command, args []string) {
-			if output == "" && lang == "" {
-				fatal("specify at least one of --output or --lang.")
+			if output == "" && lang == "" && preset == "" {
+				fatal("specify at least one of --output, --lang or --preset.")
 			}
 			appID := args[0]
 
-			if lang == "" {
+			if preset != "" {
+				switch strings.ToLower(preset) {
+				case "nextjs", "next.js":
+					lang = string(codegen.LangTypeScript)
+					tsOptions = &daemonpb.GenClientRequest_TypeScriptOptions{
+						Namespaces: false,
+						Swr:        true,
```

Comment:
I'm not sold on coupling swr with next.js. Yes, swr is also developed by the same authors but I believe React Query is increasingly becoming the library of choice for this sort of thing.

What's more, the `useBlah` helpers are only valid for endpoints that fetch data, not for mutations, since they regularly fire multiple times in the background. Unintentionally calling a stateful, mutating endpoint when not intending to is recipe for disaster, so unless we have a good way of representing which endpoints actually do that and which don't we shouldn't generate such wrappers I don't think.